src/Lean/Parser/Extra.lean

```diff
@@ -58,7 +58,7 @@ attribute [runBuiltinParserAttributeHooks]
   withPosition $ many (checkColGe "irrelevant" >> p)
 
 @[runBuiltinParserAttributeHooks, inline] def sepByIndent (p : Parser) (sep : String) (psep : Parser := symbol sep) (allowTrailingSep : Bool := false) : Parser :=
-  withPosition $ sepBy (checkColGe "irrelevant" >> p) sep psep allowTrailingSep
+  withPosition $ sepBy (checkColGe "irrelevant" >> p) sep (psep <|> checkLinebreakBefore >> pushNone) allowTrailingSep
```

Comment:
The same suggestion also applies to `sepBy1Indent` below.

Reply:
done